Automate weekly sample refresh runs

quazi-h opened the pull request:
## Why
We need to run the DAP sample refresh on a weekly schedule.
We should have a pipeline to only run sample extract and transform and write outfiles using that transform output.
Upstream changes have already ensured that the transform_records solid returns a DAP_survey_type so we do not need to modify or alter write_outfiles to handle running a single pipeline type.

We MAY need to make changes in order to pass in an additional parameter (--firecloud) to the tsv_convert script so that the correct primary key is used in the final TSV.

[Relevant ticket](https://broadinstitute.atlassian.net/browse/dspdc-1867)

## This PR
Added a new pipeline to exclusively process sample data.
Created a weekly_schedule config for the new sample pipeline.
Added the new pipelines to repositories.
Fixing incorrect RedCap token pull into env variables.
Added a new test for the new sample refresh pipeline.

## Checklist
- [ ] Documentation has been updated as needed.

quazi-h committed:
Added a new pipeline to exclusively process sample data.
Created a weekly_schedule config for the new sample pipeline.
Added the new pipelines to repositories.
Fixing incorrect RedCap token pull into env variables.
Added a new test for the new sample refresh pipeline.

aherbst-broad commented on orchestration/dap_orchestration/pipelines.py:
```diff
@@ -75,3 +77,44 @@ def refresh_data_all() -> None:
     write_outfiles(collected_outputs)
+
+
+@pipeline(
```
I don't think we need a new pipeline. Dagster let's us run subsets of a pipeline via the `solid_selection` arg to this decorator. In this case we can give it `["sample_extract_records*"]`, which will run the "sample_extract_records and all solids downstream of it (i.e., sample_extract_records -> sample_transformation_records -> write_outfiles)

aherbst-broad commented on orchestration/dap_orchestration/pipelines.py:
```diff
@@ -75,3 +77,44 @@ def refresh_data_all() -> None:
+    start_date=datetime(2021, 9, 9),
+    execution_time=time(15, 00),
+    execution_timezone="US/Eastern",
+    mode="test_mode",
```
I think the mode name is actually "test"

aherbst-broad commented on orchestration/project/build.properties:
```diff
@@ -0,0 +1 @@
+sbt.version=1.3.7
```
Is this on purpose?

quazi-broad replied:
I did not explicitly change this - should I back this out?

aherbst-broad replied:
Yeah, please do.

quazi-broad replied:
leaving this at 1.3.7 as per our discussion

quazi-h committed:
Extended dagster repositories to include base, dev, and prod.
Helmfile changes to load the correct repositories file when rendering the chart.
Removing the new pipeline and moving the weekly_sample_refresh to the prod repositories code.

aherbst-broad committed:
Update helmfile

quazi-h committed:
"testing scheduled runs"

quazi-h committed:
Merge branch 'master' into qh-dspdc-1867-weekly-dap-sample

quazi-h committed:
changing the schedule for a prod test run

quazi-h committed:
Removing an init that is no longer needed.
Mypy, formatting, pep8 changes.
Removing the sample run from test_pipelines since we removed that as a pipeline.
Formatting the date that is provided to the end_time parameter with a timezone (java offsetdatetime).

quazi-h committed:
Switching to a non-partition based scheduler for simplicity.

aherbst-broad committed:
Update (#197)

* Add test and consume schedule execution context

* Add ignore

aherbst-broad commented on ops/helmfile/dagster/apply.sh:
```diff
@@ -62,4 +62,4 @@ else
 
 helmfile --interactive $COMMAND
-fire_slack_deployment_notification ${ENV} ${GIT_SHORTHASH}
+#fire_slack_deployment_notification ${ENV} ${GIT_SHORTHASH}
```
Uncomment this.

aherbst-broad commented on orchestration/workspace.yaml:
```diff
@@ -1,2 +1,2 @@
 load_from:
-  - python_file: repositories.py
+  - python_file: prod_repositories.py
```
Point this at dev_repositories.py

quazi-h committed:
Uncommenting slack notifications (testing).
Updating cron schedule.
Pointing workspace to dev repositories.
